meterfs/test_meterfs_allocate.c

```diff
@@ -0,0 +1,124 @@
+}
+
+
+TEST_TEAR_DOWN(meterfs_allocate)
+{
+	TEST_ASSERT_EQUAL(0, test_fileEraseAll());
+}
+
+
+/* test case of allocating file bigger than flash size */
+TEST(meterfs_allocate, big_file)
+{
+	test_common.flashsz = 4 * 1024 * 1024;
+	test_common.sectorsz = 4096;
+	test_common.recordsz = 20;
+
+	test_common.sectors = ((test_common.flashsz + test_common.sectorsz) / test_common.sectorsz) + 1;
+	test_common.filesz = test_common.flashsz + test_common.sectorsz;
+
+	TEST_ASSERT_LESS_THAN(0, test_fileAllocate("file0", test_common.sectors, test_common.filesz, test_common.recordsz));
+}
+
+
+/* test case of allocating more files than allowed */
+TEST(meterfs_allocate, many_files)
+{
+	int i;
+	char fileName[9];
+
+	for (i = 0; i < (255 + 10); ++i) {
+		sprintf(fileName, "file%d", i);
+		if (i < 255)
+			TEST_ASSERT_EQUAL_MESSAGE(0, test_fileAllocate(fileName, 2, 2000, 20), fileName);
+		else
+			TEST_ASSERT_LESS_THAN_MESSAGE(0, test_fileAllocate(fileName, 2, 2000, 20), fileName);
+		memset(fileName, 0, sizeof(fileName));
+	}
+}
+
+
+/* test case of allocating files with not allowed name length */
+TEST(meterfs_allocate, file_name_len)
+{
+	TEST_ASSERT_LESS_THAN(0, test_fileAllocate("file01234", 2, 2000, 20));
+	TEST_ASSERT_EQUAL(0, test_fileAllocate("file0123", 2, 2000, 20));
+	TEST_ASSERT_LESS_THAN(0, test_fileAllocate("", 2, 2000, 20));
+}
+
+
+/* test case of allocating file with records bigger than flash sector size */
+TEST(meterfs_allocate, big_record)
+{
+	TEST_ASSERT_EQUAL(0, test_fileAllocate("file0", 6, 20000, 5000));
+}
+
+
+/* test case of allocating files with various initialization arguments */
+TEST(meterfs_allocate, var_init_args)
+{
+	TEST_ASSERT_LESS_THAN(0, test_fileAllocate("file0", 0, 0, 0));
+	TEST_ASSERT_LESS_THAN(0, test_fileAllocate("file1", 0, 2000, 20));
+	TEST_ASSERT_LESS_THAN(0, test_fileAllocate("file2", 1, 2000, 20));
+	TEST_ASSERT_LESS_THAN(0, test_fileAllocate("file3", 2, 20, 200));
+	TEST_ASSERT_LESS_THAN(0, test_fileAllocate("file4", 3, 2000000, 20));
+	TEST_ASSERT_LESS_THAN(0, test_fileAllocate("file5", 7, 2000, 0));
+
+	TEST_ASSERT_EQUAL(0, test_fileAllocate("file6", 4, 20, 20));	
+	TEST_ASSERT_EQUAL(0, test_fileAllocate("file7", 6, 2000, 20));
+	TEST_ASSERT_EQUAL(0, test_fileAllocate("file8", 8, 10, 5));
+	TEST_ASSERT_EQUAL(0, test_fileAllocate("file9", 12, 10, 5));
+	TEST_ASSERT_EQUAL(0, test_fileAllocate("file10", 10, 2000, 10));
+	TEST_ASSERT_EQUAL(0, test_fileAllocate("file11", 9, 2000, 20));
+}
+
+
+TEST_GROUP_RUNNER(meterfs_allocate)
+{
+	RUN_TEST_CASE(meterfs_allocate, big_file);
+	RUN_TEST_CASE(meterfs_allocate, many_files);
+	RUN_TEST_CASE(meterfs_allocate, file_name_len);
+	RUN_TEST_CASE(meterfs_allocate, big_record);
+	RUN_TEST_CASE(meterfs_allocate, var_init_args);
+}
+
+
+void runner(void)
+{
+	RUN_TEST_GROUP(meterfs_allocate);
+}
+
+
+int main(int argc, char *argv[])
+{
+	test_fileInit();
+	TEST_ASSERT_EQUAL(0, test_fileEraseAll());
+
+	UnityMain(argc, (const char**)argv, runner);
```

Comment:
It's not clear for me why we need explicit cast here. 

char*[] shouldn't be casted fine implicitly to const char**?